crypto/core_namemap.c

```diff
@@ -245,38 +245,31 @@ static int namemap_add_name(OSSL_NAMEMAP *namemap, int number,
                             const char *name)
 {
     int ret;
-    HT_VALUE *val = NULL;
+    HT_VALUE val = { 0 };
```

Comment:
You can do this, but theres helper macros to make this more readable.  See the IMPLEMENT_HT_VALUE_TYPE_FNS macro.  It works similarly to the DEFINE_LHASH_OF_EX macro in that it creates wrapper functions around ossl_ht_insert/delete/lookup/etc with proper typing for the types of data you want to maniuplate.  for instance, by doing this:

```
HT_START_KEY_DEFN(namekey)
HT_DEF_KEY_CHAR_ARRAY(name, 64)
HT_END_KEY_DEFN(NAMEKEY)

IMPLEMENT_HT_VALUE_TYPE_FNS(NAMENUM_ENTRY, namemap, static)

```

You can on insert do:
```
NAMENUM_ENTRY *new = malloc(...)
new->num = num;
HT_INIT_KEY(&key);
HT_SET_STRING_CASE(&key, name, name);

ossl_ht_namemap_NAMENUM_ENTRY_insert(ht, TO_HT_KEY(&key), new, NULL);
```

The advantage here being that the wrapper is properly typed to the NAMENUM_ENTRY data type, and it dynamically records the type so on lookup, you can call:

```
NAMENUM_ENTRY *e = ossl_ht_namemap_NAMENUM_ENTRY_get(map, TO_HT_KEY(&key), &v);
```
And the wrapper will automatically validate that the data type you have looked up matches the data type inserted for that key.

Reply:
IMO the type safety layer is not useful here as I am not using the hashtable for different data types.

Reply:
but you could, and (referencing above), it allows for the elimination of the stack structure, and its required lock.  It also makes for more readable and type safe code

Reply:
No, you cannot eliminate the stack structure without losing some properties of the namemap. And I do not see the better readability here. If you insist on using it, I will, but I disagree that it is making the code cleaner.

Reply:
I don't insist, I really just wanted to point out that it might help out here.